bindata/etcd/pod.yaml

```diff
@@ -10,6 +10,55 @@ metadata:
     revision: "REVISION"
 spec:
   initContainers:
+    - name: etcd-ensure-env-vars
+      image: ${IMAGE}
+      imagePullPolicy: IfNotPresent
+      terminationMessagePolicy: FallbackToLogsOnError
+      command:
+        - /bin/sh
+        - -c
+        - |
+          #!/bin/sh
+          set -euo pipefail
+
+          : "${NODE_NODE_ENVVAR_NAME_ETCD_URL_HOST?not set}"
+          : "${NODE_NODE_ENVVAR_NAME_ETCD_NAME?not set}"
+          : "${NODE_NODE_ENVVAR_NAME_IP?not set}"
+
+          if [ "${NODE_NODE_ENVVAR_NAME_IP}" != "${POD_IP}" ]; then
+            # echo the error message to stderr
+            >&2 echo "Expected node IP to be ${NODE_IP} got ${NODE_NODE_ENVVAR_NAME_IP}";
+            exit 1;
+          fi
+
+          if [ "${NODE_NODE_ENVVAR_NAME_ETCD_URL_HOST}" != "${NODE_NAME}" ]; then
```

Comment:
Is this correct? I thought NODE_NODE_ENVVAR_NAME_ETCD_URL_HOST is now going to be an IP address, and not NODE_NAME. On my cluster now, I see IP address for the URL_HOST value.

Reply:
goood catch, updated